### PR TITLE
ProxyTaskGroup in DAG

### PR DESCRIPTION
**Pull request by anuchandy**
1. Code clean-up.
2. Support for adding post-run dependents: that enables execution of subset of dependents after the execution of dependency task group.
3. Support for InvocationContext
4. More test coverage (22 tests for DAG)

### CODE REVIEW

**Comment by anuchandy** [comment]
@jianghaolu thinking of taking this for v1.4, can you please review?

File: azure-mgmt-resources/src/test/java/com/microsoft/azure/management/resources/fluentcore/dag/DAGErrorTests.java [hunk 1]
```diff
@@ -362,6 +363,8 @@ public IPancake call(Indexable indexable) {
             }
         }).toBlocking().last();
 
+        System.out.println("Seen            : " + seen);
```
Comment (jianghaolu) [inline azure-mgmt-resources/src/test/java/com/microsoft/azure/management/resources/fluentcore/dag/DAGErrorTests.java]: Remove

**Comment by anuchandy** [comment]
@martinsawicki FYI as discussed yesterday i ran the parallel tests in live mode with the new TaskGroup (this PR) and everything is working as expected.

**Comment by martinsawicki** [comment]
rock on!

**Pull request merged**